src/garage/tf/algos/rl2.py

```diff
@@ -0,0 +1,316 @@
+"""RL^2: Fast Reinforcement learning via slow reinforcement learning.
+
+Reference: https://arxiv.org/pdf/1611.02779.pdf.
+"""
+from dowel import logger, tabular
+import numpy as np
+
+from garage.misc import tensor_utils as np_tensor_utils
+from garage.np.algos import RLAlgorithm
+
+
+class RL2(RLAlgorithm):
+    """RL^2.
+
+    Args:
+        inner_algo (garage.np.algos.RLAlgorithm): Inner algorithm.
+        max_path_length (int): Maximum length for trajectories with respect
+            to RL^2. Notice that it is different from the maximum path length
+            for the inner algorithm.
+
+    """
+
+    def __init__(self, inner_algo, max_path_length):
+        self._inner_algo = inner_algo
+        self._max_path_length = max_path_length
+        self._env_spec = inner_algo.env_spec
+        self._flatten_input = inner_algo.flatten_input
+        self._policy = inner_algo.policy
+        self._discount = inner_algo.discount
+
+    def train(self, runner):
```

Comment:
AFAICT, this method is identical to `garage.tf.BatchPolopt.train`. I guess it would be weird to inherit from that though, and it would also be weird to call just one method from it.